### PR TITLE
feat: Adds StreamingProcessor for FDv2 to sdk-server package.

### PR DESCRIPTION
**Pull request by tanderson-ld**
**Requirements**

- [x] I have added test coverage for new or changed functionality

- [x] I have followed the repository's [pull request submission guidelines](../blob/main/CONTRIBUTING.md#submitting-pull-requests)
- [x] I have validated my changes against all supported platform versions

**Related issues**

SDK-849

**Describe the solution you've provided**

Inserted PayloadReader between EventSource and DataSourceUpdates.

Contract test glue code can be found on `ta/sdk-849/fdv2-streaming-datasource-contract-test-glue`

### CODE REVIEW

File: packages/shared/common/src/internal/fdv2/payloadReader.ts [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+  id: string;
+  version: number;
+  state: string;
+  basis: boolean;
+  updates: Update[];
+}
+
+export type PayloadListener = (payload: Payload) => void;
+
+export class PayloadReader {
```
Comment (tanderson-ld) [inline packages/shared/common/src/internal/fdv2/payloadReader.ts:33]: TODO: commenting

**Review by kinyoklion — CHANGES_REQUESTED** [review]

File: packages/shared/common/__tests__/internal/fdv2/PayloadReader.test.ts [hunk 2]
```diff
@@ -0,0 +1,316 @@
+import { EventListener, EventName, LDLogger } from '../../../src/api';
+import { EventStream, Payload, PayloadReader } from '../../../src/internal/fdv2/payloadReader';
+
+class MockEventStreamm implements EventStream {
```
[the remark this answers is not in the document]
Comment (tanderson-ld) [reply packages/shared/common/__tests__/internal/fdv2/PayloadReader.test.ts]: Done

File: packages/shared/common/__tests__/internal/fdv2/PayloadReader.test.ts [hunk 3]
```diff
@@ -0,0 +1,316 @@
...
+  private _listeners: {
+    [event: EventName]: EventListener;
```
[the remark this answers is not in the document]
Comment (tanderson-ld) [reply packages/shared/common/__tests__/internal/fdv2/PayloadReader.test.ts]: Done

File: packages/shared/common/src/internal/fdv2/payloadReader.ts [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+  private _processObj(kind: string, jsonObj: any): any {
+    return this._jsonObjConverters[kind]?.(jsonObj);
+  }
+
+  private _processServerIntent = (event?: { data?: ServerIntentData }) => {
+    // clear state in prep for handling data
+    this._resetState();
+
+    // if there's no payloads, return
+    if ((event?.data?.payloads.length ?? 0) <= 0) {
```
[the remark this answers is not in the document]
Comment (tanderson-ld) [reply packages/shared/common/src/internal/fdv2/payloadReader.ts]: Done

File: packages/shared/common/src/internal/fdv2/payloadReader.ts [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+  version: number;
+  state: string;
+  basis: boolean;
+  updates: Update[];
+}
+
+export type PayloadListener = (payload: Payload) => void;
+
+export class PayloadReader {
+  listeners: PayloadListener[] = [];
```
[the remark this answers is not in the document]
Comment (tanderson-ld) [reply packages/shared/common/src/internal/fdv2/payloadReader.ts]: Good call.  Done.

**Review by kinyoklion — APPROVED** [review]

**Pull request merged**